account_invoice_show_currency_rate/models/account_move.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    currency_rate_amount = fields.Float(
+        string="Rate amount", compute="_compute_currency_rate_amount", digits=0,
+    )
+    show_currency_rate_amount = fields.Boolean(
+        compute="_compute_show_currency_rate_amount", readonly=True
+    )
+
+    @api.depends(
+        "state", "date", "line_ids.amount_currency", "company_id", "currency_id"
+    )
+    def _compute_currency_rate_amount(self):
+        """ It's necessary to define value according to some cases:
+        - Case A: Currency is equal to company currency (Value = 1)
+        - Case B: Move exist previously (posted) and get real rate according to lines
+        - Case C: Get expected rate (according to date) to show some value in creation.
+        """
+        for item in self:
+            item.currency_rate_amount = 1
+            if item.state == "posted" and item.line_ids.filtered(
+                lambda x: x.amount_currency > 0
+            ):
+                amount_currency_positive = sum(
+                    item.line_ids.filtered(lambda x: x.amount_currency > 0).mapped(
+                        "amount_currency"
+                    )
+                )
+                total_debit = sum(item.line_ids.mapped("debit"))
+                item.currency_rate_amount = amount_currency_positive / total_debit
+            elif (
+                not item.line_ids.filtered(lambda x: x.amount_currency > 0)
+                or item.state != "posted"
+            ) and item.currency_id != item.company_id.currency_id:
```

Comment:
You end up repeating this condition here and above for the first `if`. To avoid it, you can convert this into:

```suggestion
           else:
               if item.currency_id != item.company_id.currency_id:
```

or you could store that info in a variable at the beginning to avoid having to make multiple calls